src/data/finset/powerset.lean

```diff
@@ -67,8 +67,71 @@ begin
+⟨(λ h t hs, h t (mem_powerset.2 hs)), (λ h _ _, h _ _)⟩
+
+/-- A version of `finset.decidable_exists_of_decidable_subsets` with a non-dependent `p`.
+Typeclass inference cannot find `hu` here, so this is not an instance. -/
+def decidable_exists_of_decidable_subsets' {s : finset α} {p : finset α → Prop}
+  (hu : Π t (h : t ⊆ s), decidable (p t)) : decidable (∃ t (h : t ⊆ s), p t) :=
+@finset.decidable_exists_of_decidable_subsets _ _ _ hu
+
+/-- A version of `finset.decidable_forall_of_decidable_subsets` with a non-dependent `p`.
+Typeclass inference cannot find `hu` here, so this is not an instance. -/
+def decidable_forall_of_decidable_subsets' {s : finset α} {p : finset α → Prop}
+  (hu : Π t (h : t ⊆ s), decidable (p t)) : decidable (∀ t (h : t ⊆ s), p t) :=
+@finset.decidable_forall_of_decidable_subsets _ _ _ hu
+
 end powerset
 
+section ssubsets
+variables [decidable_eq α]
+
+/-- For `s` a finset, `s.ssubsets` is the finset comprising strict subsets of `s`. -/
+def ssubsets (s : finset α) : finset (finset α) :=
+erase (powerset s) s
+
+@[simp] lemma mem_ssubsets {s t : finset α} : t ∈ s.ssubsets ↔ t ⊂ s :=
+by rw [ssubsets, mem_erase, mem_powerset, ssubset_iff_subset_ne, and.comm]
+
+lemma empty_mem_ssubsets {s : finset α} (h : s.nonempty) : ∅ ∈ s.ssubsets :=
+by { rw [mem_ssubsets, ssubset_iff_subset_ne], exact ⟨empty_subset s, h.ne_empty.symm⟩, }
+
+/-- For predicate `p` decidable on ssubsets, it is decidable whether `p` holds for any ssubset. -/
+instance decidable_exists_of_decidable_ssubsets {s : finset α} {p : Π t ⊂ s, Prop}
+  [hu : Π t (h : t ⊂ s), decidable (p t h)] : decidable (∃ t h, p t h) :=
+decidable_of_iff (∃ t (hs : t ∈ s.ssubsets), p t (mem_ssubsets.1 hs))
+⟨(λ ⟨t, _, hp⟩, ⟨t, _, hp⟩), (λ ⟨t, hs, hp⟩, ⟨t, mem_ssubsets.2 hs, hp⟩)⟩
+
+/-- For predicate `p` decidable on ssubsets, it is decidable whether `p` holds for every ssubset. -/
+instance decidable_forall_of_decidable_ssubsets {s : finset α} {p : Π t ⊂ s, Prop}
+  [hu : Π t (h : t ⊂ s), decidable (p t h)] : decidable (∀ t h, p t h) :=
+decidable_of_iff (∀ t (h : t ∈ s.ssubsets), p t (mem_ssubsets.1 h))
+⟨(λ h t hs, h t (mem_ssubsets.2 hs)), (λ h _ _, h _ _)⟩
+
+/-- A version of `finset.decidable_exists_of_decidable_ssubsets` with a non-dependent `p`.
+Typeclass inference cannot find `hu` here, so this is not an instance. -/
+def decidable_exists_of_decidable_ssubsets' {s : finset α} {p : finset α → Prop}
+  (hu : Π t (h : t ⊂ s), decidable (p t)) : decidable (∃ t (h : t ⊂ s), p t) :=
+@finset.decidable_exists_of_decidable_ssubsets _ _ _ _ hu
+
+/-- A version of `finset.decidable_forall_of_decidable_ssubsets` with a non-dependent `p`.
+Typeclass inference cannot find `hu` here, so this is not an instance. -/
+def decidable_forall_of_decidable_ssubsets' {s : finset α} {p : finset α → Prop}
+  (hu : Π t (h : t ⊂ s), decidable (p t)) : decidable (∀ t (h : t ⊂ s), p t) :=
+@finset.decidable_forall_of_decidable_ssubsets _ _ _ _ hu
```

Comment:
Just to check  - do these new definitions work for your actual use-case?

Reply:
Thanks yes they're fine! FWIW here's the use case https://github.com/agjftucker/exists-unique/blob/e7f795bd957befb97b8647c2cde2479bd0cd3e32/src/q_def.lean#L102. I'm aware I probably shouldn't be messing with decidable at all :) But it wasn't the only purpose of this PR, the def of `ssubsets` is used elsewhere.